File: 2190-most-frequent-number-following-key-in-an-array/2190-most-frequent-number-following-key-in-an-array.cpp
class Solution {
public:
    int mostFrequent(vector<int>& nums, int key) {
        int freq[1001] = {0};
        //count freq of all elements which are targets
        for(int i=1;i<nums.size();i++) {
            if(nums[i-1]==key) {
                freq[nums[i]]++;
            }
        }
        //find element with max frequency in freq array
        int max = 0;
        for(int i=1;i<1001;i++) {
            if(freq[i]>freq[max]) {
                max = i;
            }
        }
        return max;
    }
};

File: 2190-most-frequent-number-following-key-in-an-array/2190-most-frequent-number-following-key-in-an-array.java
class Solution {
    public int mostFrequent(int[] nums, int key) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length-1;i++) {
            if(nums[i]==key) {
                map.put(nums[i+1], map.getOrDefault(nums[i+1], 0)+1);
            }
        }
        int target = 0;
        int max = -1;
        for(Map.Entry<Integer, Integer> e: map.entrySet()) {
            if(e.getValue() > max) {
                max = e.getValue();
                target = e.getKey();
            }
        }
        return target;
    }
}